### ISSUE TITLE
The AndroidNavigationBar color (navigationBarColor) is ignored when using a Modal

### ISSUE THREAD

**Issue by RubenO88** [wix/react-native-navigation#1694]
### Issue Description

If you use a modal and pass the navigationBarColor in the navigatorStyle props it will apply the color on the screen under the modal and the modals AndroidNavigationBar remains the default color(black). This also applies when using the static navigatorStyle in the modal screen or the setStyle option.

#### General navigation controller
```javascript
....
export const navigatorStyle = {
    statusBarColor: COLORS.secondaryColor, //android only
    statusBarTextColorScheme: 'light',

    navigationBarColor: COLORS.secondaryColor, //(back/home/recent) buttonsbar on android
    navBarBackgroundColor: COLORS.primaryColor,
    navBarTextColor: 'white',
    navBarButtonColor: 'white',
    navBarNoBorder: true,
    topBarElevationShadowEnabled: false,
    navBarHideOnScroll: false,
    navBarTitleTextCentered: true
};

export function showModalWithNavBar(screen, title, data, navStyle) {
    Navigation.showModal({
        screen,
        title,
        navigatorStyle: {
            ...navigatorStyle,
            ...navStyle
        },
        appStyle: tabsStyle,
        passProps: data
    });
}
....
```
#### Tabbased screen with a nav button
```
import { push, showModalWithNavBar } from '....path to general nav controller';
.....
onNavigatorEvent(event) {
        if (event.type === 'NavBarButtonPress') {
            if (event.id === 'search') {
                this.navToSearch();
            }
        }
    }

navToSearch() {
        showModalWithNavBar(
            'searchScreen',
            null,
            null,
            {navigationBarColor: 'red'}
        );
    }

---
### Environment
* React Native Navigation version: 1.1.193
* React Native version: 0.46.1
* Platform: Android
* Device info: Device HTC one M9/Motorola Nexus 6

**Comment by RubenO88** [comment]
Thanks, @guyca 
Does your fix also work for lightbox's and if the keyboard comes up?
If not I might create another issues for those use cases.

**Comment by guyca** [comment]
Hey @RubenO88 
I think it was available in v1.1.196 or v1.1.195
Please open a ticket if this issue happens with LightBox as well
